src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -959,8 +801,9 @@ public boolean isHasRaidBeenCalculated()
      */
     public void markDirty()
     {
+        this.writeToNBT(new NBTTagCompound());
```

Comment:
You don't want this. You only want to mark dirty, so that multiple changes in the same tick (or refresh period if we don't check every tick) will be written together. This will write this to NBT multiple times.

Reply:
Yeah every 5 seconds is definitely enough, if someone places a building and crashes the world within 5 seconds, not much will be lost anyway.